### PR TITLE
Deleting a customrun on the basis of name

### PR DESCRIPTION
**Pull request by Senjuti256**
Added the delete.go and delete_test.go files for customruns to handle deletion of customruns based on name only.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
Add tkn customrun delete command.
```

### CODE REVIEW

**Comment by Senjuti256** [comment]
@piyush-garg

File: pkg/cmd/customrun/delete.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
+// Copyright © 2023 The Tekton Authors.
```
Comment (vinamra28) [inline pkg/cmd/customrun/delete.go:1]:
```suggestion
// Copyright © 2024 The Tekton Authors.
```

File: pkg/cmd/customrun/delete.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		err := deleteSingleCustomRun(clients, namespace, crName)
+		if err != nil {
+			fmt.Fprintf(s.Err, "failed to delete CustomRun %s: %v\n", crName, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func deleteSingleCustomRun(cs *cli.Clients, namespace, crName string) error {
```
Comment (vinamra28) [inline pkg/cmd/customrun/delete.go:83]: should we rename this to `deleteCustomRun` ?

**Review by vinamra28 — COMMENTED** [review]
/ok-to-test

File: pkg/cmd/customrun/delete.go (hunk 2 above)
Comment (Senjuti256) [reply pkg/cmd/customrun/delete.go]: @vinamra28 made the modifications as per your suggestion

File: pkg/cmd/customrun/delete_test.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
+package customrun
```
Comment (piyush-garg) [inline pkg/cmd/customrun/delete_test.go:1]: add header

File: pkg/cmd/customrun/delete.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			}
+
+			namespace := p.Namespace()
+
+			err = deleteCustomRuns(s, cs, namespace, crNames)
+			if err != nil {
+				return err
+			}
+
+			for _, crName := range crNames {
```
Comment (piyush-garg) [inline pkg/cmd/customrun/delete.go:62]: this should be after delete command, if err is not nil

File: pkg/cmd/customrun/delete.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		if err != nil {
+			fmt.Fprintf(s.Err, "failed to delete CustomRun %s: %v\n", crName, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func deleteCustomRun(cs *cli.Clients, namespace, crName string) error {
+	err := cs.Dynamic.Resource(customrunGroupResource).Namespace(namespace).Delete(context.TODO(), crName, metav1.DeleteOptions{})
```
Comment (piyush-garg) [inline pkg/cmd/customrun/delete.go:86]: should we do a check also first whether it exists?

**Review by vinamra28 — APPROVED** [review]

**Pull request merged**